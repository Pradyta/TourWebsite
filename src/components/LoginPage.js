import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import "../style.css";


const LoginPage = () => {
  const [username, setUsername] = React.useState("");
  const [password, setPassword] = React.useState("");
  const navigate = useNavigate();

  const handleLogin = () => {
    let result = fetch("http://localhost:5000/login", {
      method: "POST",
      body: JSON.stringify({ username, password }),
      headers: {
        "Content-Type": "applicantion/json",
      },
    })
    .then((resp) => resp.json())
    .then((data) => {
      if (data.name) {
        // console.log(data);
        localStorage.setItem("users", JSON.stringify(data));
        navigate("/");
        // alert("Welcome to Trip Advisor " + username);
        
      } else {
        alert("Please enter correct details");
      }

    })
  };
  return (
    <>
      
      <div className="relative flex flex-col justify-center min-h-screen overflow-hidden">
        <div className="card w-full p-6 m-auto bg-white rounded-md lg:max-w-xl">
          <h1 className="text-3xl font-semibold text-center text-purple-700 underline">
            Login Here
          </h1>
          <form className="mt-6">
            <div className="mb-2">
              <label
                htmlFor="username"
                className="block text-sm font-semibold text-gray-800"
              >
                User Name
              </label>
              <input
                type="text"
                value={username}
                onChange={(e) => setUsername(e.target.value)}
                placeholder="Enter username"
                className="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
              />
            </div>

            <div className="mb-2">
              <label
                htmlFor="password"
                className="block text-sm font-semibold text-gray-800"
              >
                Password
              </label>
              <input
                type="password"
                value={password}
                placeholder="Password"
                onChange={(e) => setPassword(e.target.value)}
                className="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
              />
            </div>

            <div className="mt-6">
              <button type="button"
                className="w-full px-4 py-2 tracking-wide text-white transition-colors duration-200 transform bg-purple-700 rounded-md hover:bg-purple-600 focus:outline-none focus:bg-purple-600"
                onClick={handleLogin}
              >
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
    </>
  );
};

export default LoginPage;
