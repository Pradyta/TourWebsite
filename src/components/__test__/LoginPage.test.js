import { render, screen } from '@testing-library/react';
import LoginPage from '../LoginPage';
import {BrowserRouter as Router} from 'react-router-dom';
import userEvent from "@testing-library/user-event";


describe("Test the Login Component", () => {
    test('should render Login compoennt', () => {
    const {asFragment} = render( 
    <Router>
        <LoginPage />,
    </Router>
    );
    expect(asFragment(<LoginPage/>)).toMatchSnapshot();
    });


    test("render the login form submit button on the screen", async () => {
      render(<Router>
        <LoginPage />,
    </Router>);
      const buttonList = await screen.findAllByRole("button");
      expect(buttonList).toHaveLength(1);
    });

    // test("should display alert if error", () => {
    //     render(<Router>
    //         <LoginPage />,
    //     </Router>);

    //     const username = screen.getByPlaceholderText("Enter username");
    //     const password = screen.getByPlaceholderText("Password");
    //     const buttonList = screen.getAllByRole("button");
    
    //     userEvent.type(username, "User");
    //     userEvent.type(password, "123456");
    //     userEvent.click(buttonList[0]);
    //     const error = screen.getByText("Username is not valid");
    //     expect(error).toBeInTheDocument();
    //   });

    test("should be able to submit the form", () => {
        const component = render(<Router>
            <LoginPage />,
        </Router>);
        const username = screen.getByPlaceholderText("Enter username");
        const password = screen.getByPlaceholderText("Password");
        const btnList = screen.getAllByRole("button");
    
        userEvent.type(username, "User");
        userEvent.type(password, "123456");
        userEvent.click(btnList[0]);
    
        const user = screen.getByText("User");
        expect(user).toBeInTheDocument();
      });
    });