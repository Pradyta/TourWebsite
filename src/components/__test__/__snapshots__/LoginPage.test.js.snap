// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Login Component should render Login compoennt 1`] = `
<DocumentFragment>
  <div
    class="relative flex flex-col justify-center min-h-screen overflow-hidden"
  >
    <div
      class="w-full p-6 m-auto bg-white rounded-md shadow-md lg:max-w-xl"
    >
      <h1
        class="text-3xl font-semibold text-center text-purple-700 underline"
      >
        Login Here
      </h1>
      <form
        class="mt-6"
      >
        <div
          class="mb-2"
        >
          <label
            class="block text-sm font-semibold text-gray-800"
            for="username"
          >
            User Name
          </label>
          <input
            class="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            placeholder="Enter username"
            type="text"
            value=""
          />
        </div>
        <div
          class="mb-2"
        >
          <label
            class="block text-sm font-semibold text-gray-800"
            for="password"
          >
            Password
          </label>
          <input
            class="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <div
          class="mt-6"
        >
          <button
            class="w-full px-4 py-2 tracking-wide text-white transition-colors duration-200 transform bg-purple-700 rounded-md hover:bg-purple-600 focus:outline-none focus:bg-purple-600"
            type="button"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
  ,
</DocumentFragment>
`;

exports[`should render Login compoennt 1`] = `
<DocumentFragment>
  <div
    class="relative flex flex-col justify-center min-h-screen overflow-hidden"
  >
    <div
      class="w-full p-6 m-auto bg-white rounded-md shadow-md lg:max-w-xl"
    >
      <h1
        class="text-3xl font-semibold text-center text-purple-700 underline"
      >
        Login Here
      </h1>
      <form
        class="mt-6"
      >
        <div
          class="mb-2"
        >
          <label
            class="block text-sm font-semibold text-gray-800"
            for="username"
          >
            User Name
          </label>
          <input
            class="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            type="text"
            value=""
          />
        </div>
        <div
          class="mb-2"
        >
          <label
            class="block text-sm font-semibold text-gray-800"
            for="password"
          >
            Password
          </label>
          <input
            class="block w-full px-4 py-2 mt-2 text-purple-700 bg-white border rounded-md focus:border-purple-400 focus:ring-purple-300 focus:outline-none focus:ring focus:ring-opacity-40"
            type="password"
            value=""
          />
        </div>
        <div
          class="mt-6"
        >
          <button
            class="w-full px-4 py-2 tracking-wide text-white transition-colors duration-200 transform bg-purple-700 rounded-md hover:bg-purple-600 focus:outline-none focus:bg-purple-600"
            type="button"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
  ,
</DocumentFragment>
`;
